blocks/filter.py

```diff
@@ -120,14 +121,32 @@ def __init__(self, roles=None, bricks=None, each_role=False, name=None,
         self.theano_name_regex = theano_name_regex
         self.applications = applications
 
-    def __call__(self, variables):
+    def __call__(self, variables, graph=False):
         """Filter the given variables.
 
         Parameters
         ----------
-        variables : list of :class:`~tensor.TensorVariable`
+        variables : :class:`~tensor.TensorVariable` or iterable
+            If an instance of :class:`~tensor.TensorVariable`,
+            operate on a :class:`~blocks.graph.ComputationGraph`
+            built from that variable. If an iterable (e.g. a list,
+            or :class:`~blocks.graph.ComputationGraph`) containing
+            :class:`~tensor.TensorVariable`s, operate on those variables.
+        graph : bool, optional
+            If `True`, build a `~blocks.graph.ComputationGraph` instance
+            with `variables` as the output. Default: `False`. Note that
+            `False` has no effect if `variables` is a
+            :class:`~tensor.TensorVariable` (i.e. the result is the same,
+            an instance of :class:`~blocks.graph.ComputationGraph` is
+            built).
```

Comment:
Can you give an example from the issue? It might be hard to see a use case for this argument without an explanation.